### PR TITLE
Upgrade docker-compose to 2.15

### PR DESCRIPTION
**Pull request by pasimko**
x86 is updated, powerpc still needs a build.

### CODE REVIEW

File: attributes/compose.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
-default['osl-docker']['compose']['version'] = '1.18.0'
+default['osl-docker']['compose']['version'] = '2.15'
 default['osl-docker']['compose']['filename'] = "docker-compose-Linux-#{node['kernel']['machine']}"
 case node['kernel']['machine']
 when 'x86_64'
-  default['osl-docker']['compose']['checksum'] = 'b2f2c3834107f526b1d9cc8d8e0bdd132c6f1495b036a32cbc61b5288d2e2a01'
+  default['osl-docker']['compose']['checksum'] = 'ba481d45be2b137a2a185abd05f61d6d7766dbedfa038f16e4705760767a206e'
   default['osl-docker']['compose']['url_base'] = 'https://github.com/docker/compose/releases/download'
```
Comment (ramereth) [inline attributes/compose.rb:6]: Please add `aarch64` binaries too.

**Review by ramereth — CHANGES_REQUESTED** [review]
Please also fix/update the tests

File: attributes/compose.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 when 'ppc64le'
   default['osl-docker']['compose']['checksum'] = '4458de249ca5f776738e44a6af2f869c1186b2d018dd15705045aa01a45e50c5'
```
Comment (ramereth) [inline attributes/compose.rb:8]: I just built a new version and pushed it up, please update the checksum to `2b57f69c438fadaa88dc383d8df8b98955a2ab9262c7ca2524102c8199c8efb4`

**Review by ramereth — CHANGES_REQUESTED** [review]
Please update the InSpec test.

File: test/integration/compose/inspec/compose_spec.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
 describe command('/usr/local/bin/docker-compose version') do
   its('exit_status') { should eq 0 }
-  its('stdout') { should match(/docker-compose version 1\.18\.0/) }
+  its('stdout') { should match(/docker-compose version 2\.15\.0/) }
```
Comment (ramereth) [inline test/integration/compose/inspec/compose_spec.rb:3]: You need to fix the test to do the following.
```suggestion
  its('stdout') { should match(/docker-compose version v2\.15\.0/) }
```

**Review by ramereth — CHANGES_REQUESTED** [review]
You need to fix this following line: https://github.com/osuosl-cookbooks/osl-docker/blob/7cbd57c67b6b8e45818d719b85965fead549193f/recipes/compose.rb#L23

That line should be changed to the following:
```
source "#{compose['url_base']}/v#{compose['version']}/#{compose['filename']}"
```

**Comment by pasimko** [comment]
```type=AVC msg=audit(1674242743.553:214): avc:  denied  { append } for  pid=5090 comm="newaliases" path="socket:[22505]" dev="sockfs" ino=22505 scontext=system_u:system_r:system_mail_t:s0 tcontext=system_u:system_r:init_t:s0 tclass=unix_stream_socket permissive=0```
SELinux fails to install for compose-centos-stream-8 due to OOM, and when it does converge the SELinux integration test finds this error message.

**Comment by ramereth** [comment]
@pasimko try increasing the size of the instance first to work around the OOM issue

**Comment by ramereth** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v4.7.0.  Have a nice day!

**Pull request merged**